internal/scheduler/yunikorn/scheduler.go

```diff
@@ -0,0 +1,148 @@
+package yunikorn
+
+import (
+	"encoding/json"
+	"fmt"
+
+	v1 "k8s.io/api/core/v1"
+
+	"github.com/kubeflow/spark-operator/api/v1beta2"
+	"github.com/kubeflow/spark-operator/internal/scheduler"
+	"github.com/kubeflow/spark-operator/internal/scheduler/yunikorn/resourceusage"
+)
+
+const (
+	SchedulerName = "yunikorn"
+
+	// The names are set to match the Yunikorn gang scheduling example for Spark, but these can be any
+	// value as long as what's on the pod matches the task group definition
+	// https://yunikorn.apache.org/docs/next/user_guide/gang_scheduling/#enable-gang-scheduling-for-spark-jobs
+	driverTaskGroupName   = "spark-driver"
+	executorTaskGroupName = "spark-executor"
```

Comment:
I see that in Volcano scheduler, the PodGroup names are different between SparkApplications. So I am wondering whether we need to set distinct task group names for different SparkApplications?

Reply:
For Yunikorn, a task group is a property of an application, so the task group names only need to be unique within the application they're a part of. 

Application IDs must be unique in the same way that a Volcano PodGroup name has to be, however there's a nice integration in Yunikorn to use the `spark-app-selector` label for the application ID.